base/src/main/java/org/aya/distill/ConcreteDistiller.java

```diff
@@ -153,7 +154,23 @@ yield visitCalls(false, fn, (nc, l) -> l.toDoc(options), outer,
         Doc.join(Doc.symbol("|"), idiom.barredApps().view()
           .map(app -> term(Outer.Free, app)))
       );
-      case Expr.Do aDo -> throw new UnsupportedOperationException("TODO");
+      case Expr.Do doExpr -> {
+        var doBlockDoc = doExpr.binds().map(bind -> {
+          if (bind.var() == LocalVar.IGNORED) {
+            return term(Outer.Free, bind.expr());
+          } else {
+            return visitDoBind(bind);
```

Comment:
We can omit these braces by `doExpr.binds().map(bind -> bind.var() == LocalVar.IGNORED ? term(Outer.Free, bind.expr()) : visitDoBind(bind));`